pkg/controller/controller_ref_manager.go

```diff
@@ -0,0 +1,168 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klabels "k8s.io/apimachinery/pkg/labels"
+	kschema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	kutilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/record"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	kcontroller "k8s.io/kubernetes/pkg/controller"
+)
+
+// RSControlInterface is an interface that knows how to add or delete
+// ReplicationControllers, as well as increment or decrement them. It is used
+// by the DeploymentConfig controller to ease testing of actions that it takes.
+type RCControlInterface interface {
+	PatchReplicationController(namespace, name string, data []byte) error
+}
+
+// RealRCControl is the default implementation of RCControlInterface.
+type RealRCControl struct {
+	KubeClient kclientset.Interface
```

Comment:
ok, I think for many of our controllers we have initializers, that is why I asked for one :-) But if this is copying the upstream pattern i'm ok with it.